### PR TITLE
Separate exiv2 app source files from exiv2lib src folder

### PR DESCRIPTION
**Pull request by piponazo**
For a long time I had in mind to address this task. In every C++ software project in which I have collaborated I always have separated the source files belonging to the libraries and to applications (either command line applications, GUI applications or test runners). In exiv2 we had an anomaly in that regard ... the **exiv2lib** (the C++ library) and **exiv2** cmd application were intermixed in the same folder (**src**).

Due to this situation we were carrying out some bad practices. For example, the `utils.hpp` and `utils.cpp` which were part of the library, were being recompiled again together with the exiv2 app and some sample applications. After noticing this, I decided to attack this defect by using the following strategy:

1. I firstly added some characterisation tests for some of the free functions at `utils.hpp`
2. I replaced one by one all the functions in `utils.hpp` by simpler implementations (mostly by making use of `std::filesystem`)
3. Eventually the functions at `utils.cpp` were so simple that I could just replace them in their consumer and remove the functions from `utils`.
4. After removing all these functions, the characterisation tests could be also removed.

I would recommend the reviewers to review this PR commit by commit, otherwise it might be a bit challenging to see the logic in the changes.

I noticed that we can simplify much more the code related with the files & directories thanks to the use of `std::filesystem`. However, I will keep such additional work for a different PR.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -99,39 +99,38 @@ if( EXIV2_BUILD_FUZZ_TESTS )
...
-    get_directory_property(SAMPLES DIRECTORY samples DEFINITION APPLICATIONS)
-
-    if (Python3_Interpreter_FOUND)
-        add_test(NAME bashTests
-            WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/tests
-            COMMAND cmake -E env EXIV2_BINDIR=${CMAKE_RUNTIME_OUTPUT_DIRECTORY} ${Python3_EXECUTABLE} runner.py --verbose bash_tests
-        )
-    endif()
-endif()
+if(EXIV2_BUILD_EXIV2_COMMAND)
```
Comment (piponazo) [inline CMakeLists.txt:102]: I also noticed that before we were adding the python tests to CTEST even when `EXIV2_BUILD_EXIV2_COMMAND` was OFF 😅

File: src/epsimage.cpp [hunk 2]
```diff
@@ -105,6 +105,11 @@ namespace {
     // closing part of all valid XMP trailers
     const std::string xmpTrailerEnd = "?>";
 
+    bool startsWith(const std::string& s, const std::string& start)
```
Comment (neheb) [inline src/epsimage.cpp:106]: You could replace the parameters with std::string_view. This would also allow making it constezpr and reduce binary size.

Comment (piponazo) [reply src/epsimage.cpp:106]: Totally right. I still do not have the habit of using string_view. Thanks to having reviews from you, I hope to gain that habit 😉

Comment (neheb) [reply src/epsimage.cpp:106]: Note that constexpr also implies inline. In Gerbera, I've replaced all .find() == 0 usages with startswith. The function is done a bit differently though: https://github.com/gerbera/gerbera/blob/master/src/util/tools.h#L62

File: src/sonymn_int.cpp [hunk 3]
```diff
@@ -905,7 +904,7 @@ namespace Exiv2 {
             // Ranges of models that do not support this tag
             std::string model = pos->toString();
             for (auto& m : { "DSC-", "Stellar" }) {
-                if (Util::startsWith(model, m)) {
```
Comment (kevinbackhouse) [inline src/sonymn_int.cpp:908]: I've lost track of whether we're on C++20 yet. If we are then we could use [`std::string::starts_with`](https://en.cppreference.com/w/cpp/string/basic_string/starts_with).

**Review by kevinbackhouse — APPROVED** [review]

**Pull request merged**